File: 739-daily-temperatures/daily-temperatures.java
class Solution {
    public int[] dailyTemperatures(int[] temperatures) {
        int[] days = new int[temperatures.length];
        Stack<Integer> st = new Stack<>();
        for(int i=temperatures.length - 1; i>=0; i--){
            while(!st.isEmpty() && temperatures[st.peek()]<= temperatures[i]){
                st.pop();
            }
            if(!st.isEmpty()){
                days[i] = st.peek() - i;
            }
            st.push(i);
        }
        return days;
    }
}

File: 739-daily-temperatures/daily-temperatures.py
class Solution:
    def dailyTemperatures(self, temperatures: List[int]) -> List[int]:
        n = len(temperatures)
        res = [0] * n
        st = []

        for i, temp in enumerate(temperatures):
            while st and temp > st[-1][0]:
                top = st.pop()
                res[top[1]] = i - top[1]

            st.append((temp, i))

        return res
